src/lxc/freezer.c

```diff
@@ -32,45 +32,55 @@
 #include <sys/param.h>
 
 #include "error.h"
-#include "state.h"
-#include "monitor.h"
 #include "log.h"
 #include "lxc.h"
+#include "monitor.h"
+#include "parse.h"
+#include "state.h"
 
 lxc_log_define(lxc_freezer, lxc);
 
 lxc_state_t freezer_state(const char *name, const char *lxcpath)
 {
+	int ret;
 	char v[100];
-	if (lxc_cgroup_get("freezer.state", v, 100, name, lxcpath) < 0)
+
+	ret = lxc_cgroup_get("freezer.state", v, 100, name, lxcpath);
+	if (ret < 0)
 		return -1;
 
-	if (v[strlen(v)-1] == '\n')
-		v[strlen(v)-1] = '\0';
+	v[lxc_char_right_gc(v, strlen(v))] = '\0';
```

Comment:
if v happens to not be null-terminated, i.e. v is 100 'a's, this will write a \0 to v[100] which is invalid, right?

(near as I can tell v will be filled in by lxc_read_from_file which will read 100 bytes, without null-terminating)

Reply:
fixed